### PR TITLE
Show suggested rooms on overview

### PR DESCRIPTION
**Pull request by gnunicorn**
on top of #2174 and #2175 this adds the feature to show the _suggested_ spaces and chats (if any given) on the overview rather than a random list. See it in action:

https://github.com/user-attachments/assets/9ee0e00f-a348-4284-b84c-2de8cc0bfbf9

_To the reviewer_:
Only that last commit ( a121c84ee296e53b77cdcc9f69bd6236212dc000 ) is relevant here, but as it builds on top of the previous work, all is in this branch at the moment...

### CODE REVIEW

File: .changes/2176-show-suggested.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- Suggested spaces and chats are preferably now shown on the overview of a space, if configured
```
Comment (gtalha07) [inline .changes/2176-show-suggested.md:1]: I had a impression that suggestions can automatically show up based on the relevant interactions of rooms and don't need to populate manually. Is this how the sdk supports it OR we intend to choose this way?

File: app/lib/common/providers/space_providers.dart [hunk 2]
```diff
@@ -353,6 +359,64 @@ final remoteChatRelationsProvider =
...
+    if (suggestedRooms.isEmpty) {
+      return (List<String>.empty(), List<SpaceHierarchyRoomInfo>.empty());
+    }
+    final toIgnore = relatedSpaces.knownChats.toList();
+    final localRooms = relatedSpaces.knownChats
+        .where((r) => suggestedRooms.contains(r))
+        .toList();
+    final roomHierarchy =
+        await ref.watch(spaceRemoteRelationsProvider(spaceId).future);
+    // filter out the known rooms
```
Comment (gtalha07) [inline app/lib/common/providers/space_providers.dart:379]: To me it feels like this should all happen on the rust-side and if possible, the provider can yield unjoined or unknown chats/spaces which are relevant to the rooms we are part of. Something worth possible for future?

File: app/lib/common/widgets/room/room_hierarchy_card.dart [hunk 3]
```diff
@@ -97,55 +92,28 @@ class ConvoHierarchyCard extends ConsumerWidget {
...
-              ),
-              RoomHierarchyOptionsMenu(
-                isSuggested: roomInfo.suggested(),
-                childId: roomId,
-                parentId: parentId,
-              ),
-            ],
-          ),
-      onTap: onTap,
+      // subtitle: subtitle,
```
Comment (gtalha07) [inline app/lib/common/widgets/room/room_hierarchy_card.dart:108]: comment ;)

File: app/lib/features/space/widgets/space_sections/spaces_section.dart [hunk 4]
```diff
@@ -43,6 +52,47 @@ class SpacesSection extends ConsumerWidget {
...
+          isShowSeeAllButton: true,
+          onTapSeeAll: () => context.pushNamed(
+            Routes.spaceRelatedSpaces.name,
+            pathParameters: {'spaceId': spaceId},
+          ),
+        ),
+        spacesListUI(
+          suggestedLocalSpaces,
+          config.listingLimit,
+          // showOptions: false,
```
Comment (gtalha07) [inline app/lib/features/space/widgets/space_sections/spaces_section.dart:81]: comment :)

**Review by gtalha07 — APPROVED** [review]
LGTM!

File: .changes/2176-show-suggested.md (hunk 1 above)
Comment (gnunicorn) [reply .changes/2176-show-suggested.md:1]: Not entirely sure I get your question. The way that "suggested" work is that it is a flag in the `child`-room event, marking specific `children` spaces as `suggested`. this is explicitly done by the parent room admin and allows them to curate the list of spaces/chats to be shown and suggest to the user...

**Comment by gnunicorn** [comment]
rebased on the main rebase, removed the commenting but other than that all is the same.

**Pull request merged**